services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/Metadata.java

```diff
@@ -0,0 +1,90 @@
+package com.mapbox.api.directions.v5.models;
+
+import androidx.annotation.Nullable;
+import com.google.auto.value.AutoValue;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.TypeAdapter;
+import com.google.gson.annotations.SerializedName;
+import com.mapbox.api.directions.v5.DirectionsAdapterFactory;
+
+import java.util.Map;
+
+/**
+ * Object representing metadata associated with a route request.
+ */
+@AutoValue
+public abstract class Metadata extends DirectionsJsonObject {
+
+  /**
+   * Create a new instance of this class by using the {@link Metadata.Builder} class.
+   *
+   * @return {@link Metadata.Builder} for creating a new instance
+   */
+  public static Builder builder() {
+    return new AutoValue_Metadata.Builder();
+  }
+
+  /**
+   * Map containing a key/value pair of all the properties inside `map` element.
+   * @return map of key/value pairs
+   */
+  @Nullable
+  @SerializedName("map")
+  public abstract Map<String, String> versionMap();
```

Comment:
metadata response does not limit this key-valued container to be **versions** only. While right now all possible values are versions, I'd expect that other non-version keys might be added also.
is there a better name to be more generic? (`mapProperties`? `mapMetadata`? `mapInfo`? having key name as `map` and java structure as `Map` does not help here)